test/suite/test_compact07.py

```diff
@@ -0,0 +1,212 @@
+    def get_size(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(all)')
+        size = stat_cursor[stat.dsrc.block_size][2]
+        stat_cursor.close()
+        return size
+    
+    def get_free_space(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, 'statistics=(all)')
+        bytes = stat_cursor[stat.dsrc.block_reuse_bytes][2]
+        stat_cursor.close()
+        return bytes // megabyte
+    
+    def get_pages_rewritten(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, None)
+        pages_rewritten = stat_cursor[stat.dsrc.btree_compact_pages_rewritten][2]
+        stat_cursor.close()
+        return pages_rewritten
+        
+    def delete_range(self, uri):
+        c = self.session.open_cursor(uri, None)
+        for r in range(self.delete_ranges_count):
+            start = r * self.table_numkv // self.delete_ranges_count
+            for i in range(self.delete_range_len):
+                c.set_key(start + i)
+                c.remove()
+        c.close()
+    
+    # Test the basic functionality of the background compaction server. 
+    def test_background_compact_usage(self):
+        # Create a small table that compact should skip over.
+        uri_small = self.uri_prefix + '_small'
+        ds = SimpleDataSet(self, uri_small, self.table_numkv // 2, key_format=self.key_format, value_format=self.value_format)
+        ds.populate()
+        self.delete_range(uri_small)
+              
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+    
+            # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+            # than one iteration.
+            self.delete_range(uri)
+
+        # Reopen the connection to force the object to disk.
+        self.reopen_conn()
+        
+        small_file_free_space = self.get_free_space(uri_small)
+        
+        # Allow background compaction to run for some time. Set a low free_space_target larger
+        # than the smaller file.
+        self.session.compact(None,f'background=true,free_space_target={small_file_free_space + 1}MB')
+        
+        # Check that the background server is running
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 1)
+        stat_cursor.close()
+        
+        # Background compaction should run through every file as listed in the metadata file.
+        # Periodically check how many files we've compacted until we compact all of them.
```

Comment:
Is it ok if only one file contributes to the counter? Or do we want to check that we are compacting different URIs?

Reply:
I wanted to check if background compaction successfully loops through all the files listed in the metadata file.

Reply:
Sorry I misread the code, what you did is correct.